election_predictor/interface/main.py

```diff
@@ -0,0 +1,245 @@
+
+    # Handle y_train results cleaning (create a fake election result)
+    # (removes actuals results, we are trying to predict)
+    y_train.loc[y_train['next_elec_date'] == election_date,
+         ['LAB_ACT', 'CON_ACT', 'LIB_ACT', 'GRE_ACT', 'BRX_ACT', 'SNP_ACT', 'UKI_ACT', 'PLC_ACT', 'OTH_ACT']] = np.nan
+
+    # Fetch mean values of each value in X_test
+    X_test_median = X_test.mean()
+
+    # Create imputation for y_train to impute over actuals we are trying to predict
+    imputation_values = {
+        'CON_ACT': X_test_median['CON_FC'],
+        'LAB_ACT': X_test_median['LAB_FC'],
+        'LIB_ACT': X_test_median['LIB_FC'],
+        'BRX_ACT': X_test_median['BRX_FC'],
+        'GRE_ACT': X_test_median['GRE_FC'],
+        'OTH_ACT': X_test_median['OTH_FC'],
+        'PLC_ACT': X_test_median['PLC_FC'],
+        'SNP_ACT': X_test_median['SNP_FC'],
+        'UKI_ACT': X_test_median['UKI_FC']
+    }
+
+    y_train = y_train.fillna(value=imputation_values)
+    y_train = y_train.fillna(value=0)
+
+    # Finally drop month_x columns from X_train and X_test
+    X_train.drop(columns=['Month_x'], inplace=True)
+    X_test.drop(columns=['Month_x'], inplace=True)
+
+
+    # Handle training and testing UK GE parties vote share
+
+    # Handle matrix conversion for train feature data
+    X_train_matrix = np.array(X_train)
+
+    parties = [
+        'LAB', 'CON', 'LIB', 'GRE', 'BRX', 'SNP', 'UKI', 'PLC',
+        'OTH'
+    ]
+
+    trained_models = { }
+
+    for party, party_code in enumerate(parties):
+        # Set party target train and test
+        party_y_train = y_train[party_code + "_ACT"]
+        party_y_test = y_test[party_code + "_ACT"]
+
+        # Handle XGBoost Regressor modelling
+        # Handle model fetch and initialisation
+        xgb_regressor = XGBoostModel()
+        xgbr_model = xgb_regressor.initialize_model()
+
+        # Handle model compiling
+        xgb_regressor.compile_model(xgbr_model) # Uses default model parameters from params.py
+
+        # Handle model training
+        trained_model = xgb_regressor.train_model(xgbr_model, X_train_matrix, party_y_train)
+
+        trained_models[party_code] = trained_model
+
+        #TODO Reintroduce model evaluation logic for RMSE
+        # # Handle model evaluation
+        # rmse_score = xgb_regressor.evaluate_model(trained_model, X_test_matrix, party_y_test).mean()
+
+        # train_test_results[party_code] = {
+        #     "rmse_score": rmse_score,
+        #     "trained_model": trained_model
+        # }
+
+    #Handle prediction (ensure input features are transformed via preprocessor instance)
+    X_predict_matrix = np.array(X_test)
+
+    election_predictions = { }
+
+    for party_code, model in trained_models.items():
+        trained_model = model
+
+        predicted_vote_share = trained_model.predict(X_predict_matrix).mean()
+
+        election_predictions[party_code] = predicted_vote_share
+
+    return election_predictions
```

Comment:
this currently only returns national vote share (without SCO correction?), where do constituency seats get predicted?

Reply:
Constituency seats logic that CK built is coming to the function soon, for initial MVP of live source data am focussing on vote share only and will add this in the coming days.